### ISSUE TITLE
GA specific Fence

### ISSUE THREAD

**Issue by abhinavvishnu** [GlobalArrays/ga#29]
NWChemEx has requested a feature in which it would be possible to execute GA_Fence, which would take a GA handle as an argument.

Feature requested by Robert Harrison

**Comment by jeffhammond** [comment]
@abhinavvishnu Makes sense.  It is certainly a useful feature.  I only mentioned NWChem because @jeffdaily suggested possibly changing the API, and I was thinking about what codes might notice it.
